src/Generation/ResourcesGenerator.php

```diff
@@ -507,4 +513,56 @@ private static function getQueryParams(MethodDetails $method): Vector
                 ->map(fn ($f) => $f->name);
         return $queryParams;
     }
+
+    /**
+     * For a given method and service, returns any fields that are available
+     * for autopopulation given the restrictions below.
+     * The field is a top-level string field of a unary method's request message.
+     * The field is not annotated with `google.api.field_behavior = REQUIRED`.
+     * The field name is listed in `google.api.publishing.method_settings.auto_populated_fields`.
+     * The field is annotated with `google.api.field_info.format = UUID4`.
+     *
+     * @return array<string, string> Fields to autopopulate in GAX if unset by user
+     */
+    private static function getUuid4FieldsToAutoPopulate(
+        MethodDetails $method,
+        ServiceDetails $serviceDetails
+    ): array {
+        $autoPopulatedFields = Vector::new([]);
+
+        $methodSettings = $serviceDetails->serviceYamlConfig->methodSettings;
+        foreach ($methodSettings as $methodSetting) {
+            $autoPopulatedFieldNames = iterator_to_array($methodSetting->getAutoPopulatedFields());
+            if (!empty($autoPopulatedFieldNames)) {
+
+                // Check if the $methodSetting is for this particular $method
+                $splitMethodName = explode('.', $methodSetting->getSelector());
+                $methodName = $splitMethodName[array_key_last($splitMethodName)];
+                if ($methodName !== $method->name) {
+                    continue;
+                }
+
+                // Process only if the method is unary
+                if (!$method->isStreaming()) {
```

Comment:
This check should be at the start of the method, exit as early as possible since we know the `$method` from the outset.

Reply:
Yep makes sense. Thanks